src/Control/Foldl/NonEmpty.hs

```diff
@@ -375,16 +601,77 @@ minimumBy cmp = Fold1 (\begin -> Fold min' begin id)
         _  -> x
 {-# INLINABLE minimumBy #-}
 
--- | Upgrade a fold to accept the 'Fold1' type
+-- | Upgrade a "non-empty" fold to accept the 'Fold1' type
```

Comment:
I think you want to escape quotes, otherwise haddock tries to interpret them as  a module reference.  Same comment for the other quotes in the documentation:

```suggestion
-- | Upgrade a \"non-empty\" fold to accept the 'Fold1' type
```